src/Native/System.Security.Cryptography.Native/CMakeLists.txt

```diff
@@ -65,7 +65,7 @@ if (APPLE)
     add_custom_command(TARGET System.Security.Cryptography.Native POST_BUILD
         COMMAND ${CMAKE_INSTALL_NAME_TOOL} -change /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib @rpath/libcrypto.1.0.0.dylib $<TARGET_FILE:System.Security.Cryptography.Native>
         COMMAND ${CMAKE_INSTALL_NAME_TOOL} -change /usr/local/opt/openssl/lib/libssl.1.0.0.dylib @rpath/libssl.1.0.0.dylib $<TARGET_FILE:System.Security.Cryptography.Native>
-        COMMAND ${CMAKE_INSTALL_NAME_TOOL} -add_rpath @loader_path $<TARGET_FILE:System.Security.Cryptography.Native>
+        COMMAND ${CMAKE_INSTALL_NAME_TOOL} -add_rpath @loader_path -add_rpath /usr/local/share/dotnet/redist $<TARGET_FILE:System.Security.Cryptography.Native>
```

Comment:
DYLD_LIBRARY_PATH and DYLD_FALLBACK_LIBRARY_PATH are both usually unset.  DYLD_LIBRARY_PATH, if unset, is treated as the empty set.  If anyone is using it, it will continue to win.

DYLD_FALLBACK_LIBRARY_PATH, if unset, is treated as $HOME/lib;/usr/local/lib;/usr/lib.

We can't add $HOME/lib (AFAIK rpath entries don't do environment substitution).  /usr/lib is under the System Integrity Protection scope, so it can't be modified.  That leaves /usr/local/lib as the only path that we're depending on right now, which is where our instructions say to put openssl.

And we currently have \@loader_path there, so that if someone does a standalone application it will use the deployed library copy.  So \@loader_path needs to beat /usr/local/lib, which needs to beat /usr/local/share/dotnet/redist.  (User local override, system copy, the fallback redist)

DYLD_\*: https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/DynamicLibraryUsageGuidelines.html#//apple_ref/doc/uid/TP40001928-SW12
SIP: https://support.apple.com/en-us/HT204899